Kudu.Core/Deployment/Generator/ExternalCommandBuilder.cs

```diff
@@ -85,5 +114,94 @@ protected void RunCommand(DeploymentContext context, string command)
+            {
+                defaultPath = System.Environment.GetEnvironmentVariable(WellKnownEnvironmentVariables.PostDeploymentActionsDirectoryKey);
+            }
+
+            if (FileSystemHelpers.DirectoryExists(defaultPath))
+            {
+                scriptFolders.Add(defaultPath);
+            }
+
+            // D:\Program Files (x86)\SiteExtensions
+            string siteExtensionFolder = Path.Combine(System.Environment.GetFolderPath(System.Environment.SpecialFolder.ProgramFilesX86), "SiteExtensions");
+            // Find all enviroment variable with format {package id}_EXTENSION_VERSION,
+            // And find scripts from PostDeploymentActions" folder
+            foreach (DictionaryEntry entry in System.Environment.GetEnvironmentVariables())
+            {
+                string key = entry.Key.ToString();
+                string val = entry.Value.ToString();
+                NuGetVersion ver = null;
+                if (key.EndsWith("_EXTENSION_VERSION", StringComparison.Ordinal)
+                    && (string.Equals("latest", val, StringComparison.OrdinalIgnoreCase) || NuGetVersion.TryParse(val, out ver)))   // either latest or a specific version
+                {
+                    if (ver != null)
+                    {
+                        // "D:\Program Files (x86)\SiteExtensions", "{PackageName}", "{version}", "PostDeploymentActions"
+                        //  => D:\Program Files (x86)\SiteExtensions\{PackageName}\{version}\PostDeploymentActions
+                        string scriptFolder = Path.Combine(siteExtensionFolder, key.Replace("_EXTENSION_VERSION", string.Empty), ver.ToNormalizedString(), PostDeploymentActions);
+                        if (FileSystemHelpers.DirectoryExists(scriptFolder))
+                        {
+                            scriptFolders.Add(scriptFolder);
+                        }
+                    }
+                    else
+                    {
+                        // Get the latest version
+                        // "D:\Program Files (x86)\SiteExtensions", "{PackageName}"
+                        //  => D:\Program Files (x86)\SiteExtensions\{PackageName}
+                        string packageRoot = Path.Combine(siteExtensionFolder, key.Replace("_EXTENSION_VERSION", string.Empty));
+                        if (FileSystemHelpers.DirectoryExists(packageRoot))
+                        {
+                            string[] allVersions = FileSystemHelpers.GetDirectories(packageRoot);   // expecting a list of version e.g D:\Program Files (x86)\SiteExtensions\{PackageName}\{version}
+                            if (allVersions.Count() > 0)
+                            {
+                                // latest version comes first
+                                Array.Sort(allVersions, (string v1, string v2) =>
+                                {
+                                    NuGetVersion ver1 = NuGetVersion.Parse(new DirectoryInfo(v1).Name);
```

Comment:
it is intend to fail deployment if we failed to parse string to semver. which mean there must be invalid site extensions installed. i thought stack trace would self described. but true that will be helpful if we log more error message. will update.